finesse/hardware/plugins/spectrometer/dummy_opus_interface.py

```diff
@@ -81,13 +81,13 @@ def connect(self) -> None:
         self._start_connecting()
         self._finish_connecting()
 
-    def cancel(self) -> None:
-        """Cancel the current measurement."""
+    def stop(self) -> None:
+        """Stop the current measurement."""
         self._cancel_measuring()
         self._reset_after_cancelling()
         logging.info("Cancelling current measurement")
 
-    def stop(self) -> None:
+    def _on_measure_finished(self) -> None:
         """Finish measurement successfully."""
         self._start_finishing_measuring()
         self._finish_measuring()
```

Comment:
As you are discussing, this is very confusing and I'm not sure I follow the use case for these two paths now that the "Cancel" button is gone. We have the following paths in the state machine:

- `Stop` path:
`measuring -> cancelling -> connected`
- `on_measure_finish` path:
`measuring -> finishing -> connected`

Why are they different? When each should be used? Regardless of being triggered by the user or not, it seems to me that the second one is to be followed under normal operating conditions, while the first one is followed in exceptional circumstances. If that's right, why if there is a timeout (line 75), we are triggering the normal operating path rather than the exceptional one?- What other actions need to be revisited based on this?